scripts/upgrade

```diff
@@ -1,76 +1,179 @@
-sudo php composer-setup.php
-sudo php -r "unlink('composer-setup.php');"
-sudo ./composer update -n --no-dev;
-sudo ./bin/roadiz generate:nsentities;
-sudo ./bin/roadiz orm:schema-tool:update --dump-sql;
-sudo ./bin/roadiz orm:schema-tool:update --force;
-sudo ./bin/roadiz cache:clear -e prod
+    sudo ./concrete/bin/concrete5 c5:update --rerun --verbose --allow-as-root
 popd
 
-# Modify Nginx configuration file and copy it to Nginx conf directory
-nginx_conf=../conf/nginx.conf
-sed -i "s@YNH_WWW_PATH@$path_url@g" $nginx_conf
-sed -i "s@YNH_WWW_ALIAS@$src_path/@g" $nginx_conf
-# If a dedicated php-fpm process is used:
-#
-# sed -i "s@YNH_WWW_APP@$app@g" $nginx_conf
-sudo cp $nginx_conf /etc/nginx/conf.d/$domain.d/$app.conf
-
-### PHP (remove if not used) ###
-# If a dedicated php-fpm process is used:
-# # Modify PHP-FPM pool configuration and copy it to the pool directory
-final_path=/var/www/$app
-sed -i "s@__NAMETOCHANGE__@$app@g" ../conf/php-fpm.conf
-sed -i "s@__FINALPATH__@$final_path@g" ../conf/php-fpm.conf
-sed -i "s@__USER__@$app@g" ../conf/php-fpm.conf
-finalphpconf=/etc/php5/fpm/pool.d/$app.conf
-sudo cp ../conf/php-fpm.conf $finalphpconf
-sudo chown root: $finalphpconf
-sudo chmod 644 $finalphpconf
-finalphpini=/etc/php5/fpm/conf.d/20-$app.ini
-sudo cp ../conf/php-fpm.ini $finalphpini
-sudo chown root: $finalphpini
-sudo chmod 644 $finalphpini
-sudo service php5-fpm restart
-### PHP end ###
-
-# If app is public, add url to SSOWat conf as skipped_uris
-if [[ $is_public -eq 1 ]]; then
-  # See install script
-  ynh_app_setting_set "$app" unprotected_uris "/"
+### Verify the checksum of a file, stored by `ynh_store_file_checksum` in the install script.
+### And create a backup of this file if the checksum is different. So the file will be backed up if the admin had modified it.
+#ynh_backup_if_checksum_is_different "$final_path/CONFIG_FILE"
+# Recalculate and store the checksum of the file for the next upgrade.
+#ynh_store_file_checksum "$final_path/CONFIG_FILE"
+
+#=================================================
+# SETUP LOGROTATE
+#=================================================
+ynh_print_info "Upgrading logrotate configuration..."
+
+# Use logrotate to manage app-specific logfile(s)
+#ynh_use_logrotate --non-append
+
+#=================================================
+# SETUP SYSTEMD
+#=================================================
+ynh_print_info "Upgrading systemd configuration..."
+
+# Create a dedicated systemd config
+#ynh_add_systemd_config
+
+#=================================================
+# GENERIC FINALIZATION
+#=================================================
+# SECURE FILES AND DIRECTORIES
+#=================================================
+
+# Set permissions on app files
+chown -R root: $final_path
```

Comment:
In the others scripts you do `chown $app: -R` !?

Reply:
I corrected it

Reply:
You have replace it by `chown -R www-data: $final_path`
As you use a specific user for php-fpm, that user will run the php process and handle files.
Which means with www-data your user will have no permission to write into the directory. Which is a good thing if the app doesn't need it. But if so, keep root to be sure only root can write.

Reply:
Sorry, I made that change, launch a test and gone at work ^^ Haven't seen that it didn't worked